test/IntentEngineTest.py

```diff
@@ -58,3 +58,63 @@ def testSelectBestIntent(self):
         intent = next(self.engine.determine_intent(utterance))
         assert intent
         assert intent['intent_type'] == 'Parser2'
+
+    def testDropIntent(self):
+        parser1 = IntentBuilder("Parser1").require("Entity1").build()
+        self.engine.register_intent_parser(parser1)
+        self.engine.register_entity("tree", "Entity1")
+
+        parser2 = (IntentBuilder("Parser2").require("Entity1")
+                   .require("Entity2").build())
+        self.engine.register_intent_parser(parser2)
+        self.engine.register_entity("house", "Entity2")
+
+        utterance = "go to the tree house"
+
+        intent = next(self.engine.determine_intent(utterance))
+        assert intent
+        assert intent['intent_type'] == 'Parser2'
+
+        assert self.engine.drop_intent_parser('Parser2') is True
+        intent = next(self.engine.determine_intent(utterance))
+        assert intent
+        assert intent['intent_type'] == 'Parser1'
+
+    def testDropEntity(self):
+        parser1 = IntentBuilder("Parser1").require("Entity1").build()
+        self.engine.register_intent_parser(parser1)
+        self.engine.register_entity("laboratory", "Entity1")
+        self.engine.register_entity("lab", "Entity1")
+
+        utterance = "get out of my lab"
+        utterance2 = "get out of my laboratory"
+        intent = next(self.engine.determine_intent(utterance))
+        assert intent
+        assert intent['intent_type'] == 'Parser1'
+
+        intent = next(self.engine.determine_intent(utterance2))
+        assert intent
+        assert intent['intent_type'] == 'Parser1'
+
+        # Remove Entity and re-register laboratory and make sure only that
+        # matches.
+        self.engine.drop_entity(query='Entity1')
+        self.engine.register_entity("laboratory", "Entity1")
+
+        # Sentence containing lab should not produce any results
+        with self.assertRaises(StopIteration):
+            intent = next(self.engine.determine_intent(utterance))
+
+        # But sentence with laboratory should
+        intent = next(self.engine.determine_intent(utterance2))
+        assert intent
+        assert intent['intent_type'] == 'Parser1'
+
+    def testDropRegexEntity(self):
+        self.engine.register_regex_entity(r"the dog (?P<Dog>.*)")
+        self.engine.register_regex_entity(r"the cat (?P<Cat>.*)")
+        assert len(self.engine._regex_strings) == 2
+        assert len(self.engine.regular_expressions_entities) == 2
```

Comment:
Now that I'm seeing the usage in the tests, `query` seems like an ambiguous parameter name. Perhaps we should just use `entity_type` like we do for `register_entity`?

Reply:
Fixed this